apps/web/src/actions/evaluations/updateContent.ts

```diff
@@ -11,19 +15,53 @@ export const updateEvaluationContentAction = authProcedure
   .input(
     z.object({
       id: z.number(),
-      metadata: z.object({
-        prompt: z.string(),
-      }),
+      metadata: z
+        .discriminatedUnion('type', [
+          z.object({
+            type: z.literal(EvaluationMetadataType.LlmAsJudgeAdvanced),
+            prompt: z.string(),
+          }),
+          z.object({
+            type: z.literal(EvaluationMetadataType.LlmAsJudgeSimple),
+            providerApiKeyId: z.number().optional(),
+            model: z.string().optional(),
+            objective: z.string().optional(),
+            additionalInstructions: z.string().optional(),
+          }),
+        ])
+        .optional(),
+      configuration: z
+        .discriminatedUnion('type', [
+          z.object({
+            type: z.literal(EvaluationResultableType.Boolean),
+            falseValueDescription: z.string().optional(),
+            trueValueDescription: z.string().optional(),
+          }),
+          z.object({
+            type: z.literal(EvaluationResultableType.Number),
+            minValue: z.number().optional(),
+            maxValue: z.number().optional(),
+            minValueDescription: z.string().optional(),
+            maxValueDescription: z.string().optional(),
+          }),
+          z.object({
+            type: z.literal(EvaluationResultableType.Text),
+            valueDescription: z.string(),
+          }),
+        ])
```

Comment:
This can be dry `configuration: ConfigurationType,` and use in both actions

Reply:
i don't get the comment sorry

Reply:
The enums are the same when creating and updating. Extract it to be reused in both.